src/fromager/read.py

```diff
@@ -0,0 +1,22 @@
+import io
+import pathlib
+import typing
+from contextlib import contextmanager
+
+from .request_session import session
+
+
+@contextmanager
+def read_file_or_url(
+    path_or_url: str | pathlib.Path,
+) -> typing.Generator[io.TextIOWrapper, typing.Any, None]:
+    location = str(path_or_url)
+    if location.startswith("http://") or location.startswith("https://"):
```

Comment:
Let's also support `file://`. 
```suggestion
    if location.startswith(("https://", "http://", "file://")):
```

Several `str` methods support `str | tuple[str, ...]`.

```
Help on method_descriptor:

startswith(...) unbound builtins.str method
    S.startswith(prefix[, start[, end]]) -> bool

    Return True if S starts with the specified prefix, False otherwise.
    With optional start, test S beginning at that position.
    With optional end, stop comparing S at that position.
    prefix can also be a tuple of strings to try.
```